### PR TITLE
Book request approved by admin

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: app/controllers/admin/requests_controller.rb [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+      flash[:success] = t ".success"
+      redirect_to admin_requests_path
+    else
+      redirect_to admin_requests_path
+    end
+  end
+
+  private
+  def load_request
+    @request = Request.find params[:id]
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: cancancan, you don't use it?

File: app/models/request.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Request < ActiveRecord::Base
   belongs_to :user
   enum status: [:pending, :approved]
+  scope :pending_list, ->{ where status: nil }
```
Comment (dieunb) [inline app/models/request.rb]: convention `{where status: nil}`, and why not `{where status: :pending}`

File: app/controllers/admin/requests_controller.rb [hunk 3]
```diff
@@ -1,5 +1,21 @@
 class Admin::RequestsController < ApplicationController
+  before_action :load_request, only: [:update]
+
   def index
-    @requests = Request.all
+    @requests = Request.pending_list
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: if you only get pending, and what about another request?

File: app/controllers/admin/requests_controller.rb [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
   end
 
   def update
-    if @request.not_approved!
-      flash[:success] = t ".not_approve"
-      redirect_to admin_requests_path
-    else
-      redirect_to admin_requests_path
+    if params[:commit] == APPROVE
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: why not use metaprogramming to update :D

File: app/models/request.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 class Request < ActiveRecord::Base
   belongs_to :user
   enum status: [:pending, :approved, :not_approved]
+  scope :pending_list, ->{ where status: 0 }
```
Comment (dieunb) [inline app/models/request.rb]: no need, pls read enum documentation again

File: app/controllers/admin/requests_controller.rb [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
   end
 
   def update
-    if @request.not_approved!
-      flash[:success] = t ".not_approve"
-      redirect_to admin_requests_path
-    else
-      redirect_to admin_requests_path
+    [:approve, :not_approve].each do |state|
+      if params[:commit] == I18n.t(".#{state}")
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: don't use i18n for params commit

Comment (Rownaknira) [inline app/controllers/admin/requests_controller.rb]: Should I use string in params commit ?
Without loop, how I use metaprogramming ??

File: app/controllers/admin/requests_controller.rb [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+        if @request.send "#{state}d!"
```
Comment (dieunb) [inline app/controllers/admin/requests_controller.rb]: simple
```
if @request.send(params[:state])
  # puts success message
else
  # puts errors message
end
```

**Pull request merged**